dspace-api/src/main/java/org/dspace/uclouvain/consumer/FillerValueRemoverConsumer.java

```diff
@@ -0,0 +1,248 @@
+                        itemService.removeMetadataValues(
+                            context,
+                            item,
+                            mvsToClear
+                        );
+                        metadataModified = true;
+                    } catch (SQLException e) {
+                        logger.warn("Could not remove unwanted metadata values from item with id: " + uuid, e);
+                    }
+                }
+
+                for (MetadataValue mv: mvsToReplace) {
+                    try {
+                        // Replace the given metadata with a placeholder.
+                        itemService.replaceMetadata(
+                            context,
+                            item,
+                            mv.getSchema(),
+                            mv.getElement(),
+                            mv.getQualifier(),
+                            mv.getLanguage(),
+                            CrisConstants.PLACEHOLDER_PARENT_METADATA_VALUE,
+                            mv.getAuthority(),
+                            mv.getConfidence(),
+                            mv.getPlace()
+                        );
+                        metadataModified = true;
+                    } catch (SQLException e) {
+                        logger.warn(
+                            "Could not replace with placeholder for metadata "
+                            + mv.getMetadataField() + " of item: " + uuid,
+                            e
+                        );
+                    }
+                }
+
+                if (metadataModified) {
+                    itemService.update(context, item);
+                }
+            } catch (Exception e) {
+                logger.error("An error occurred while cleaning filler values for item: [" + uuid + "]", e);
+            }
+        }
+        itemToProcess.clear();
+    }
+
+    /**
+     * Given a specific inputSet, extract all DCInput that are in a form group in the given list.
+     * 
+     * @param stepConfig The current submission step.
+     * @param inputSet The set containing all the DCInput.
+     * @param groupFields The list to fill with DCInput.
+     * @param dciReader The reader used to get the inputs that are inside a given form group.
+     * @param isFormGroup A flag to indicate if we are processing a form group.
+     * @throws DCInputsReaderException If we could not get the desired form group.
+     */
+    private void extractFormGroupFields(
+        SubmissionStepConfig stepConfig,
+        DCInputSet inputSet,
+        List<String> groupFields,
+        DCInputsReader dciReader,
+        boolean isFormGroup
+    ) throws DCInputsReaderException {
+        for (DCInput[] fields: inputSet.getFields()) {
+            for (DCInput input: fields) {
+                // If the input has a 'group-like' type, search for 'children' fields and process them.
+                if (StringUtils.equalsAny(input.getInputType().toLowerCase(),
```

Comment:
`input.getInputType()` could return null.... strange but it's true. So you need to test '!=null' before using 'tolowercase`